### ISSUE TITLE
GVExport v2.1.23 has a small bug

### ISSUE THREAD

**Issue by xiaoyang2013** [Neriderc/GVExport#547]
When the administrator is not logged in, when entering the individual's GVExport from the menu bar at the top of the individual page, the xref individual marked by the current URL will not be located, but when the administrator account is logged in, it can be displayed normally in the "Include anyone related to" column.
2.1.21 does not have this problem

**Comment by Neriderc** [comment]
Thanks for reporting this! I can confirm that I see the same, it works when logged in but not when logged out.

I'll try to find some time to dig into this soon, and will include it in the 2.2.3 release whenever that comes.

**Comment by xiaoyang2013** [comment]
Can I also update it in v2.1? I don't have PHP 8.3 or above, so I can't install webtress 2.2.

**Comment by xiaoyang2013** [comment]
Can I also update it in v2.1? I don't have PHP 8.3 or above, so I can't install webtress 2.2.

**Comment by Neriderc** [comment]
Last week I released a GVExport 2.2.2 that should support webtrees 2.1. I took some ideas from other modules and created a helper function that checks the version and uses different methods depending on the version. It should now support both webtrees 2.1 and 2.2.

**Comment by xiaoyang2013** [comment]
![asfdasd](https://github.com/user-attachments/assets/6fc8bbee-a24b-4e17-976c-4f753459713c)
Hello, is there any other place I need to change? I changed it on "GVExport v2.1.23", but it didn't take effect.

**Comment by Neriderc** [comment]
No, that's the change. Did you also check the setting:
![treatment of source individuals](https://github.com/user-attachments/assets/b42767aa-959e-48a8-8399-210cb33e45a4)

I added this on top of 2.2.2, but I can't see any changes between 2.1.23 and 2.2.2 that would affect this.

**Comment by xiaoyang2013** [comment]
Hi, I installed GVExport 2.2.2 on webtress2.1 today, but the problem still exists. When not logged in, the subject in the URL is displayed first, and then it automatically switches to the default subject in about 1 second.
![111](https://github.com/user-attachments/assets/7b418a1c-a3b3-47a4-8701-06ae21461231)

![2222](https://github.com/user-attachments/assets/12954011-db1b-413c-9721-7d9fc96d268a)

**Comment by Neriderc** [comment]
Could you try two things for me?

First, can you update from the main branch? Here on GitHub, go to the "Code" tab (the main page). At the top left, can you check it says "main", then click the big green "Code" button and choose "Download zip".

![screen shot of above](https://github.com/user-attachments/assets/827b651a-d135-4c3c-a480-f14d3030de50)

Unzip and install this into the modules_v4 as you normally install. This should make sure we are both running the exact same version.

Secondly, can you hard-reload the page? Normally it's enough to hold Shift while you click the reload button in your browser or press Ctrl+Shift+R, but you can also use the option in your browser settings to clear the cache.

Also, what browser are you using?

**Comment by xiaoyang2013** [comment]
Yes, I confirmed that I installed it this way: download the master branch, then delete all old GVExports, and then copy the complete GVExport package.

I tried two browsers: Google Chrome 109 and Firefox 115. I cleared the cache and cookies in both browsers.

**Comment by xiaoyang2013** [comment]
The file size of the package I downloaded is: 2.06 MB (2,165,034 bytes)

**Comment by xiaoyang2013** [comment]
This is what I saw in the background

![sdf](https://github.com/user-attachments/assets/bac56cba-4d4b-48ee-a3c8-8424eee6cc2c)

**Comment by Neriderc** [comment]
I've identified a potential race condition, possibly causing things to happen out of order. I'll work on getting a new version for you to test, but I've found another issue I'm trying to resolve and it's taking a bit of time.

**Comment by Neriderc** [comment]
I've just merged another pull request that I hope will address the issue. Can you update from the main branch again and see if it's working?

**Comment by xiaoyang2013** [comment]
Yes, I see the problem has been solved, and I am very grateful for your contribution.
In addition, as shown in the figure below, the search button is displayed on the left. Is this a change made in the new version?
![asdfasdf](https://github.com/user-attachments/assets/5fe698c3-a58e-428a-a283-1cbaa5b4aac9)
